examples/esp32-with-cred/esp32-server-with-cred.rs

```diff
@@ -0,0 +1,227 @@
+                max_fds: 5,
+            })
+        })?;
+    }
+
+    #[cfg(feature = "qemu")]
+    let (ip, _block_eth) = {
+        use esp_idf_hal::prelude::Peripherals;
+        info!("creating eth object");
+        let mut eth = Box::new(esp_idf_svc::eth::EspEth::wrap(
+            esp_idf_svc::eth::EthDriver::new_openeth(
+                Peripherals::take().unwrap().mac,
+                sys_loop_stack.clone(),
+            )
+            .unwrap(),
+        )?);
+        let _ = eth_configure(&sys_loop_stack, &mut eth)?;
+        let ip = Ipv4Addr::new(10, 1, 12, 187);
+        (ip, eth)
+    };
+
+    #[cfg(not(feature = "qemu"))]
+    info!("load vars...");
+    let nvs_vars = NvsStaticVars::new()?;
+
+    info!("starting wifi...");
+    #[allow(clippy::redundant_clone)]
+    #[cfg(not(feature = "qemu"))]
+    let (ip, _wifi) = {
+        let wifi = start_wifi(
+            periph.modem,
+            sys_loop_stack,
+            &nvs_vars.wifi_ssid,
+            &nvs_vars.wifi_pwd,
+        )?;
+        (wifi.wifi().sta_netif().get_ip_info()?.ip, wifi)
+    };
+
+    let webrtc_certificate = WebRtcCertificate::new(
+        nvs_vars.robot_dtls_cert,
+        nvs_vars.robot_dtls_key_pair,
+        &nvs_vars.robot_dtls_cert_fp,
+    );
+
+    let cert: [Vec<u8>; 2] = [nvs_vars.robot_srv_pem_chain, nvs_vars.robot_srv_pem_ca];
+    let key = nvs_vars.robot_srv_der_key;
+    let tls_cfg = Esp32TlsServerConfig::new(cert, key.as_ptr(), key.len() as u32);
+
+    let cfg = AppClientConfig::new(nvs_vars.robot_secret, nvs_vars.robot_id, ip, "".to_owned());
+
+    serve_web(cfg, tls_cfg, repr, ip, webrtc_certificate);
+    Ok(())
+}
+
+#[cfg(feature = "qemu")]
+fn eth_configure<'d, T>(
+    sl_stack: &EspSystemEventLoop,
+    eth: &mut esp_idf_svc::eth::EspEth<'d, T>,
+) -> anyhow::Result<Ipv4Addr> {
+    let mut eth = esp_idf_svc::eth::BlockingEth::wrap(eth, sl_stack.clone())?;
+    eth.start()?;
+    let ip_info = eth.eth().netif().get_ip_info()?;
+
+    info!("ETH IP {:?}", ip_info.ip);
+    Ok(ip_info.ip)
+}
+
+#[cfg(not(feature = "qemu"))]
+fn start_wifi(
+    modem: impl Peripheral<P = esp_idf_hal::modem::Modem> + 'static,
+    sl_stack: EspSystemEventLoop,
+    ssid: &str,
+    password: &str,
+) -> anyhow::Result<Box<BlockingWifi<EspWifi<'static>>>> {
+    let nvs = EspDefaultNvsPartition::take()?;
+    let mut wifi = BlockingWifi::wrap(
+        EspWifi::new(modem, sl_stack.clone(), Some(nvs.clone()))?,
+        sl_stack,
+    )?;
+    let ssid_heapless = ssid.into();
+    let password_heapless = password.into();
+    let wifi_configuration = WifiConfiguration::Client(WifiClientConfiguration {
+        ssid: ssid_heapless,
+        bssid: None,
+        auth_method: AuthMethod::WPA2Personal,
+        password: password_heapless,
+        channel: None,
+    });
+    debug!("setting wifi configuration...");
+    wifi.set_configuration(&wifi_configuration)?;
+
+    wifi.start().unwrap();
```

Comment:
it looks like `start` returns a result anyway so why do we need to `unwrap` here instead of using `?`? Same for `connect` and `wait_netif_up` below.